### PR TITLE
CONTRIBUTING.md update with some more best practices.

### PR DESCRIPTION
**Pull request by jkwatson**
Adds a section about class member ordering, since it's not covered in the google style guide, seemed like a good idea to document it, so we don't need to argue about it.
Also adds a line about the instability of `toString()` methods.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,6 +67,15 @@ which uses [google-java-format](https://github.com/google/google-java-format) li
 * By default, all arguments/members are treated as non-null. Every argument/member that can be `null` must be annotated with `@Nullable`.
 * The project aims to provide a consistent experience across all the public APIs. It is important to ensure consistency (same look and feel) across different public packages.
 * Use `final` for public classes everywhere it is possible, this ensures that these classes cannot be extended when the API does not intend to offer that functionality.
+* In general, we use the following ordering of class members:
+    * static fields
```
Comment (anuraaga) [inline CONTRIBUTING.md:71]: Oh - also should we call out final before non-final?

Comment (jkwatson) [reply CONTRIBUTING.md:71]: yeah, that's probably a good call. I'll update.

Comment (jkwatson) [reply CONTRIBUTING.md]: done!

**Review by bogdandrutu — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,6 +67,15 @@ which uses [google-java-format](https://github.com/google/google-java-format) li
...
 * Use `final` for public classes everywhere it is possible, this ensures that these classes cannot be extended when the API does not intend to offer that functionality.
+* In general, we use the following ordering of class members:
+    * static fields (final before non-final)
+    * non-static fields (final before non-final)
+    * constructors
+    * static methods
+    * instance methods
+    * inner classes
+* Adding `toString()` overrides on classes is encouraged, but we only use `toString()` to provide debugging assistance. The implementations
+of all `toString()` methods should be considered to be unstable unless explicitly documented otherwise.
```
Comment (bogdandrutu) [inline CONTRIBUTING.md:78]: This is not very visible for the users of the API. Consider to add similar statement in the API readme? Or package info?

**Review by bogdandrutu — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (jkwatson) [reply CONTRIBUTING.md:78]: Agreed. Probably both in the readme and the package info. I can add that to this PR, or create a separate one, your call.

Comment (anuraaga) [reply CONTRIBUTING.md:78]: Let's do separate since the title reflects this PR well :)

**Pull request merged**